### PR TITLE
Improve create plan fields validation

### PR DESCRIPTION
**Pull request by ciremusyoka**
Part of #1124 and #1101
Disable create plan submit button if the form is not valid.

### CODE REVIEW

File: src/components/forms/PlanForm/index.tsx [hunk 1]
```diff
@@ -1061,7 +1061,7 @@ const PlanForm = (props: PlanFormProps) => {
               className="btn btn-block"
               color="primary"
               aria-label={SAVE_PLAN}
-              disabled={isSubmitting || Object.keys(errors).length > 0}
+              disabled={isSubmitting || Object.keys(errors).length > 0 || !isValid}
```
Comment (p-netm) [inline src/components/forms/PlanForm/index.tsx:1064]: @ciremusyoka question? in what way is `Object.keys(errors).length > 0` different from `!isValid`

**Comment by moshthepitt** [comment]
@ciremusyoka have you confirmed that this works to ensure that you cant create a plan with no jurisdiction?

File: src/components/forms/PlanForm/index.tsx (hunk 1 above)
Comment (ciremusyoka) [reply src/components/forms/PlanForm/index.tsx:1064]: @p-netm from my observations sometimes before any field is touched `errors` are an empty object. That is why i decided to use the two checks.

**Comment by ciremusyoka** [comment]
> @ciremusyoka have you confirmed that this works to ensure that you cant create a plan with no jurisdiction?

Yah! Currently the `Save Plan` button is disabled until the form is valid.

File: src/components/forms/PlanForm/index.tsx (hunk 1 above)
Comment (p-netm) [reply src/components/forms/PlanForm/index.tsx:1064]: :+1:

**Pull request merged**